DQM/BeamMonitor/python/BeamSpotDipServer_cff.py

```diff
@@ -0,0 +1,24 @@
+import FWCore.ParameterSet.Config as cms
+from DQMServices.Core.DQMEDAnalyzer import DQMEDAnalyzer
+
+beamSpotDipServer = DQMEDAnalyzer("BeamSpotDipServer",
+  monitorName = cms.untracked.string("BeamSpotDipServer"),
+  #
+  verbose = cms.untracked.bool(False),
+  testing = cms.untracked.bool(False),
+  #
+  subjectCMS = cms.untracked.string("dip/CMS/Tracker/BeamSpot"),
+  subjectLHC = cms.untracked.string("dip/CMS/LHC/LuminousRegion"),
+  subjectPV  = cms.untracked.string("dip/CMS/Tracker/PrimaryVertices"),
+  #
+  readFromNFS = cms.untracked.bool(False),
+  DCSStatus   = cms.untracked.string("scalersRawToDigi"),
+  #
+  sourceFile  = cms.untracked.string(
+    "/nfshome0/dqmpro/BeamMonitorDQM/BeamFitResults.txt"),
```

Comment:
Ok as I mentioned above you are reading the txt file that is filled by the "HLT workflow" ([beamhlt_dqm_sourceclient-live_cfg.py#L106](https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/beamhlt_dqm_sourceclient-live_cfg.py#L106)), while the legacy workflow ([beam_dqm_sourceclient-live_cfg.py#L96](https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py#L96)) is writing to 
`/nfshome0/dqmpro/BeamMonitorDQM/BeamFitResultsOld.txt`

We should decide which one we want to publish, or at least add an option to make this configurable.
For example for the Pilot Beam Test in October I think only the legacy workflow (which is fed by ZeroBias) will produce actual BeamSpot values.

@gennai @mmusich any thoughts?

Reply:
Hi @francescobrivio this comes a bit out of context for me. What did we use to publish before?

Reply:
Unfortunately I don't know what was published in the past, @sikler do you know? 
I will try to dig up the old DIP results.

Reply:
I have implemented two possible workflows for DIP publishing:
- read the BeamSpotOnlineObject and DCSStatus, and transmit their data (this is the default)
- and the old way of reading the /nfs files (BeamFitResults.txt and BeamFitResults_TkStatus.txt), can be selected by ('readFromNFS = True')

So, the default now is BeamSpotOnlineObject and and DCSStatus ('readFromNFS = False')

Reply:
Hello @sikler 
What is the reason or advantage of the operation mode reading the data from the condition DB (the BeamSpotOnlineObject)? Why it has been chosen as the default choice?
Also, taking into account that we have two workflows ("Legacy" and "HLT") populating two different tags, which one are you going to select?

Reply:
Hi @ggovi 
> What is the reason or advantage of the operation mode reading the data from the condition DB (the BeamSpotOnlineObject)? Why it has been chosen as the default choice?

It was felt that reading /nfs files was not a good practice, so a full online solution with BeamSpotOnlineObject was the way to go.

> Also, taking into account that we have two workflows ("Legacy" and "HLT") populating two different tags, which one are you going to select?

My understanding was that there will be some switch at the creation of BeamSpotOnlineObject, and that will decide with which beamspot would be used for online and also to be transmitted to LHC's DIP. So the beamspot used by online should match the DIP version.

Reply:
@sikler thanks for the answer. The fragility of nfs filesystem sounds indeed like a good reason for preferring the DB as input. But it has a cost - see the variables added in the payload, which will be presumably not consumed by HLT, but only "bridged out" for the DIP storage. 
Concerning the choice of the two tags, I think the actual implementation has been evolved with respect to the earlier discussion, and now we have actually the two payloads (in the two different tags), with no "DB-stored" information about which one has been  selected/consumed by hlt. @francescobrivio may comment about this...

Reply:
- @ggovi the BeamSpot online workflows do not rely on `nfs`, they write directly the payload and upload it to CondDB. `nfs` is only used to store the txt files so I don't think there is a problem there.

- @sikler regarding the choice of the two workflow it's like Giacomo wrote: both payloads are produced and uploaded to the DB in different tags. Then it's up to the consumer to choose the preferred one. At the moment there are 2 consumers: the HLT and the DQM monitoring.
   Both rely on an ESProducer ([OnlineBeamSpotESProducer.cc](https://github.com/cms-sw/cmssw/blob/master/RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc)) that reads both tags and takes the final decision on what BeamSpot to use, since they are based on the same producer the decision take is the same for both (this has been commissioned and verified during MWGR and CRUZET).
   You can see an example of the implementation in the DQM monitoring client [onlinebeammonitor_dqm_sourceclient-live_cfg.py](https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/onlinebeammonitor_dqm_sourceclient-live_cfg.py)
   
   So in the end we have to decide if what we publish in DIP is what we actually use in HLT (and in Express), or if we want to publish always the same result in any case (e.g. always the legacy BeamSpot).

Reply:
@francescobrivio well if the workflows write files outside the local node disk, they could be still affected by nfs issues. In this case, it would be unreasonable to address the issue only on the DIP workflow side.  Maybe we should discuss the overall picture. I'll try to setup a discussion.